### ISSUE TITLE
Missing aws region even when region is set in coverband settings

### ISSUE THREAD

**Issue by kbaum** [danmayer/coverband#120]
Noticed that even with region set within coverband, I get an error requiring the environment variable when running the rake coverage generation.

```
 Aws::Errors::MissingRegionError: missing region; use :region option or export region name to ENV['AWS_REGION']
/app/vendor/bundle/ruby/2.5.0/gems/aws-sdk-core-2.11.126/lib/aws-sdk-core/plugins/regional_endpoint.rb:34:in `after_initialize'
/app/vendor/bundle/ruby/2.5.0/gems/aws-sdk-core-2.11.126/lib/seahorse/client/base.rb:84:in `block in after_initialize'
/app/vendor/bundle/ruby/2.5.0/gems/aws-sdk-core-2.11.126/lib/seahorse/client/base.rb:83:in `each'
/app/vendor/bundle/ruby/2.5.0/gems/aws-sdk-core-2.11.126/lib/seahorse/client/base.rb:83:in `after_initialize'
/app/vendor/bundle/ruby/2.5.0/gems/aws-sdk-core-2.11.126/lib/seahorse/client/base.rb:21:in `initialize'
/app/vendor/bundle/ruby/2.5.0/gems/aws-sdk-core-2.11.126/lib/seahorse/client/base.rb:105:in `new'
/app/vendor/bundle/ruby/2.5.0/gems/aws-sdk-resources-2.11.126/lib/aws-sdk-resources/resource.rb:169:in `extract_client'
/app/vendor/bundle/ruby/2.5.0/gems/aws-sdk-resources-2.11.126/lib/aws-sdk-resources/resource.rb:15:in `initialize'
/app/vendor/bundle/ruby/2.5.0/bundler/gems/coverband-2cb9764f8d10/lib/coverband/s3_report_writer.rb:43:in `new'
/app/vendor/bundle/ruby/2.5.0/bundler/gems/coverband-2cb9764f8d10/lib/coverband/s3_report_writer.rb:43:in `s3'
/app/vendor/bundle/ruby/2.5.0/bundler/gems/coverband-2cb9764f8d10/lib/coverband/s3_report_writer.rb:47:in `bucket'
/app/vendor/bundle/ruby/2.5.0/bundler/gems/coverband-2cb9764f8d10/lib/coverband/s3_report_writer.rb:32:in `object'
/app/vendor/bundle/ruby/2.5.0/bundler/gems/coverband-2cb9764f8d10/lib/coverband/s3_report_writer.rb:19:in `persist!'
/app/vendor/bundle/ruby/2.5.0/bundler/gems/coverband-2cb9764f8d10/lib/coverband/reporters/simple_cov_report.rb:46:in `report'
/app/vendor/bundle/ruby/2.5.0/bundler/gems/coverband-2cb9764f8d10/lib/coverband/tasks.rb:12:in `block (2 levels) in <top (required)>'
```

```ruby
::Coverband.configure do |config|
  config.root              = Dir.pwd
  config.redis             = Redis.new
  config.ignore            = %w[vendor .erb$]
  config.startup_delay     = 10
  config.percentage        = ENV.fetch('VTS_COVERBAND_PERCENTAGE', 10.0).to_f
  config.logger            = Rails.logger
  config.verbose           = ENV['VTS_COVERBAND_VERBOSE'].present?
  config.memory_caching    = true
  # s3 bucket for recording results
  config.s3_bucket = my-bucket
  config.s3_region = 'us-east-1'
end
```

**Comment by danmayer** [comment]
Odd, I will take a look, might need to do another small point release while working on coverband 3 for this.

**Comment by danmayer** [comment]
weird I can't figure out how this would be happening... 

set on config and pulled here: https://github.com/danmayer/coverband/blob/master/lib/coverband/configuration.rb#L50

which is set as an arg and passed in here (the line referenced in your stack trace):  https://github.com/danmayer/coverband/blob/master/lib/coverband/reporters/simple_cov_report.rb#L46

which then set as options and passed in here
https://github.com/danmayer/coverband/blob/master/lib/coverband/s3_report_writer.rb#L41

which matches the docs... https://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html
I am also using this successfully on our production project to see to EU-west-1

I can't seem to reproduce in console... We do have slightly different AWS sdk versions, but unlikely to be the issue... any ideas @kbaum 

```
 aws-sdk (2.10.129)
 aws-sdk-resources (= 2.10.129)
 aws-sdk-core (2.10.129)
```

**Comment by danmayer** [comment]
hoping to fix this for the 2.0.3 release, but still can't reproduce any ideas here @kbaum or should I release 2.0.3 without it?

**Comment by danmayer** [comment]
Cool no worries, @kbaum  then I will test the various improvements pulled into 2.0.3 and try to release tonight or tomorrow.
